### PR TITLE
Create a new API to get realtime data for pdu

### PR DESCRIPTION
**Pull request by Pengpengwanga**
Signed-off-by: Pengpeng Wang <pengpengw@vmware.com>

### CODE REVIEW

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java [hunk 1]
```diff
@@ -320,6 +326,171 @@ public void insertRealtimeData(@PathVariable("id") String assetID,
...
+            break;
+         }
+      }
+      return result;
+   }
+
+   @ResponseStatus(HttpStatus.OK)
+   @RequestMapping(value = "/pdu/{id}/realtimedata", method = RequestMethod.GET)
+   public List<MetricData> getPduMetricsData(@PathVariable("id") String assetID) {
+      long starttime = System.currentTimeMillis() - TEN_MINUTES;
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java]: let's change to 5

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java [hunk 2]
```diff
@@ -320,6 +326,171 @@ public void insertRealtimeData(@PathVariable("id") String assetID,
...
+      });
+   }
+
+   public List<ValueUnit> getValueUnits(List<RealTimeData> realtimeDatas){
+      List<ValueUnit> valueunits = null;
+      if(realtimeDatas == null || realtimeDatas.isEmpty()) {
+         return valueunits;
+      }
+      valueunits = new ArrayList<>();
+      realTimeDataSortByCreateTime(realtimeDatas);
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java]: let;s consider use bubble , only fetch the largest.

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java [hunk 3]
```diff
@@ -320,6 +326,171 @@ public void insertRealtimeData(@PathVariable("id") String assetID,
...
+      }
+      for(Map.Entry<String, Map<String, String>> sensorFormularMap : sensorFormulars.entrySet()) {
+         Map<String, String> sensorLocationAndIdMap = sensorFormularMap.getValue();
+         Set<String> assetIds = new HashSet<String>();
+         for(String formular : sensorLocationAndIdMap.values()) {
+            assetIds.addAll(Arrays.asList(formular.split("\\+|-|\\*|/|\\(|\\)")));
+         }
+         switch (sensorFormularMap.getKey()) {
+         case MetricName.PDU_HUMIDITY:
+            Set<String> humidityAssetIds = metricNameAndAssetIdsMap.get(MetricName.PDU_HUMIDITY);
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java]: humidty|outlet   humidty|inlet

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java (hunk 2 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java]: Done

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java (hunk 3 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java]: done

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java [hunk 4]
```diff
@@ -320,6 +325,42 @@ public void insertRealtimeData(@PathVariable("id") String assetID,
...
+
+      Asset pdu = assetRepository.findOne(assetID);
+      //sensor metrics data, such as temperature or humidity
+      Map<String, Map<String, Map<String, String>>> formulars = pdu.getMetricsformulars();
+      Map<String, Map<String, String>> sensorFormulars = null;
+      if(formulars != null && !formulars.isEmpty()) {
+         sensorFormulars = formulars.get(FlowgateConstant.SENSOR);
+      }
+      if(sensorFormulars != null) {
+         Iterator<Map<String, String>> ite = sensorFormulars.values().iterator();
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java]: Map<assetid, Realtimedata>

Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java]: Done

**Pull request merged**